### ISSUE TITLE
Complex conjugate generates unstable hash

### ISSUE THREAD

**Issue by estebanag** [dask/dask#2341]
When running this script, the dask `name` changes on every run:
```python
import numpy as np
import dask.array as da

A = np.ones((10, 10))*5
B = np.ones((10, 10))*6
A = da.from_array(A, chunks=A.shape)
B = da.from_array(B, chunks=B.shape)
C = A + 1j * B
print(C.conj().name)  # this changes on every run
```
Shouldn't this be stable? [Here](https://github.com/dask/dask/issues/2228)'s a related issue which has already been solved.

Thanks!

**Comment by jcrist** [comment]
Hmmm, this is weird. It looks like numpy ufuncs don't have stable pickle representations. It's consistent per-run, but not consistent on each initialization of python. Issue is true for any ufunc:

```python
(dask) jcrist dask $ cat test.py
import numpy as np
import pickle

print(pickle.dumps(np.sin))
(dask) jcrist dask $ python test.py
b'\x80\x03cnumpy.core\n_ufunc_reconstruct\nq\x00X\x10\x00\x00\x00numpy.core.umathq\x01X\x03\x00\x00\x00sinq\x02\x86q\x03Rq\x04.'
(dask) jcrist dask $ python test.py
b'\x80\x03cnumpy.core\n_ufunc_reconstruct\nq\x00X\n\x00\x00\x00numpy.coreq\x01X\x03\x00\x00\x00sinq\x02\x86q\x03Rq\x04.'
```

Looking at the numpy code, this is because numpy uses the `pickle.whichmodule` function to find the module with the object, which iterates through the (non-deterministic) `sys.modules.items()` to find the first module with the object.

Fixing now.

**Comment by jcrist** [comment]
Fixed by #2343.

```bash
(dask) jcrist dask $ cat test.py
import numpy as np
import dask.array as da

A = np.ones((10, 10))*5
B = np.ones((10, 10))*6
A = da.from_array(A, chunks=A.shape)
B = da.from_array(B, chunks=B.shape)
C = A + 1j * B
print(C.conj().name)
(dask) jcrist dask $ python test.py
conjugate-c084d7def1755e7fa5a29fe632bcd088
(dask) jcrist dask $ python test.py
conjugate-c084d7def1755e7fa5a29fe632bcd088
(dask) jcrist dask $ python test.py
conjugate-c084d7def1755e7fa5a29fe632bcd088
```
